docs/source/build.md

```diff
@@ -156,6 +156,7 @@ RAFT's cmake has the following configurable flags available:.
 | BUILD_TESTS               | ON, OFF              | ON | Compile Googletests |
 | BUILD_BENCH               | ON, OFF              | OFF | Compile benchmarks |
 | raft_FIND_COMPONENTS      | compiled distributed | | Configures the optional components as a space-separated list |
+| RAFT_ENABLE_CTK_MATH_DEPENDENCIES | ON, OFF | ON | Link against CUDA Math Libraries in `raft::raft` | 
```

Comment:
We might want to note something like: “math libraries are required for (these headers/features) but optional for consumers of (these headers/features).”

Reply:
Thanks for the suggestion here @bdice. We've tried that approach in the past and it made the headers (and the core library) very hard to consume. Most users don't (and shouldn't) care about these dependencies, which is why we're making this an "opt-out" feature. To try and list out all of the namespaces/headers which do and don't use these libraries would be an arduous task and nearly impossible to maintain.

Reply:
That is totally reasonable!